spec/features/groups/manage_group_members_spec.rb

```diff
@@ -50,19 +64,51 @@ def when_i_click_add_editor
     click_link "Add an editor"
   end
 
+  def when_i_click_add_editor_or_group_admin
+    click_link "Add an editor or group admin"
+  end
+
   def then_i_should_see_the_add_editor_form
     expect(page.find("h1")).to have_text "Add an editor to this group"
     expect_page_to_have_no_axe_errors(page)
   end
 
+  def then_i_should_see_the_add_editor_or_group_admin_form
+    expect(page.find("h1")).to have_text "Add an editor or group admin to this group"
+    expect_page_to_have_no_axe_errors(page)
+  end
+
   def when_i_fill_in_the_add_editor_form
     fill_in "Enter the email address of the person you want to add", with: user3.email
     click_button "Add this person"
   end
 
-  def then_i_should_see_the_new_editor_in_the_list
+  def when_i_fill_in_the_add_group_admin_form
+    fill_in "Enter the email address of the person you want to add", with: user3.email
+    choose("Group admin")
+    click_button "Add this person"
+  end
+
+  def then_i_should_see_the_new_group_admin
+    expect(page.find("h1")).to have_text "Group 1"
+
+    user_row = page.find("td", text: user3.name).ancestor("tr")
+    expect(user_row).to have_text user3.name
+    expect(user_row).to have_text user3.email
+    expect(user_row).to have_text "Group admin"
```

Comment:
Good suggestion Al, thank you! I needed to tweak it but I think keeping the expectations in a within block makes the intent clearer regardless.

Reply:
Nice find @thomasiles 👏

Reply:
🤔 Do these end-to-end tests need to do this amount of checking? should there not be view specs that are cheaper and can mock the expected data out? I'd say e2e should just check you are on the correct page or check updated data has been updated and displayed. If it's grainualar then unit testing with views spec is the way to go.

Reply:
you are right. We were interested in checking the table row because thats how we know the data has been updated. 

We didn't want to just check that the string 'Editor' was on the page without knowing it was associated with our user. Maybe just:

```
    expect(page).to have_table(with_rows: [[user3.name, 'Editor']])
```

is enough to know that the user has been added to the group as an editor?